ale_linters/cpp/clangtidy.vim

```diff
@@ -0,0 +1,15 @@
+" Author: vdeurzen <tim@kompiler.org>
+" Description: clang-tidy linter for cpp files
+
+" Set this option to change the clang-tidy options for warnings for C.
+if !exists('g:ale_cpp_clangtidy_options')
+    let g:ale_cpp_clangtidy_options = '-std=c++14 -Wall'
+endif
+
+call ale#linter#Define('cpp', {
+\   'name': 'clangtidy',
+\   'output_stream': 'stdout',
+\   'executable': 'clang-tidy',
+\   'command': g:ale#util#stdin_wrapper . ' -- ' . g:ale_cpp_clangtidy_options,
```

Comment:
You need to pass the file extension as the first argument to `stdin_wrapper`. You will also need to add on `clang-tidy`, the actual command you are wrapping itself is missing here. Can you put `--` before other options? `--` usually marks the end of all options in GNU land.